### ISSUE TITLE
DefaultServlet behaviour wrong for `welcomeFiles`

### ISSUE THREAD

**Issue by janbartel** [eclipse/jetty.project#10608]
**Jetty version(s)**
10/11/12

The behaviour for `welcomeFiles` = `true` or `exact` does not match the javadoc.

The desired behaviour is explained in the javadoc for jetty-10:

```
 *  welcomeServlets   If true, attempt to dispatch to welcome files
 *                    that are servlets, but only after no matching static
 *                    resources could be found. If false, then a welcome
 *                    file must exist on disk. If "exact", then exact
 *                    servlet matches are supported without an existing file.
 *                    Default is false.

```

However in jetty-10, 11 and 12, that is not the behaviour - a resource must exist or we throw a `404`.

See in jetty-10 `ResourceService` around lines 240/245.
See in jetty-12 `DefaultServlet` around lines 458/462.

For example given:

``` java

        _contextHandler = new ServletContextHandler();
        _contextHandler.setContextPath("/context");
        _contextHandler.setBaseResourceAsPath("/some/path/contextResources");
        _contextHandler.setWelcomeFiles(new String[] {"index.x"});
        ServletHolder rHolder = _contextHandler.addServlet(DefaultServlet.class, "/r/*");
        rHolder.setInitParameter("welcomeServlets", "true");
        _contextHandler.addServlet(RogerThatServlet.class, "*.x");
```

A request to `/context/r/` should dispatch to `/index.x`, but will not proceed to the `RogerThatServlet` _unless_ the directory `/some/path/contextResources/r` exists, which it should not need to according to the javadoc.

**Comment by grgrzybek** [comment]
Hello - any plans about this one?

**Comment by gregw** [comment]
> Hello - any plans about this one?

Yes - we will be looking in the next week or two, as we plan to release 12.1.0 next month

**Comment by gregw** [comment]
@janbartel I can't see anything wrong in the implementation of welcome servlets.  See #12982 for improved testing and clarified documentation.
If you think there is still something wrong, can you create the test scenarios in the new unit test I created?

**Comment by janbartel** [comment]
@gregw I think your new test case at line `1930` covers the original case reported here. If that test is passing, we should be good for `ee11`. Might we worthwhile backporting this test case to `ee9`, and `ee10`.
